revocation/crl/cache/memory.go

```diff
@@ -0,0 +1,76 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+// MemoryCache is an in-memory cache that stores CRL bundles.
+//
+// The cache is built on top of the sync.Map to leverage the concurrency control
+// and atomicity of the map, so it is suitable for writing once and reading many
+// times. The CRL is stored in memory as a Bundle type.
+//
+// MemoryCache doesn't handle cache cleaning but provides the Delete and Clear
+// methods to remove the CRLs from the memory.
+type MemoryCache struct {
```

Comment:
It might be worth considering moving the cache implementation out of the core-go codebase. We could define a cache interface in the core-go module, and then allow clients/users to pick whichever cache implementation they prefer to use with the core-go functionality. This would decouple the cache logic from the core-go code, providing more flexibility for end users.

Reply:
Yes, I agree. This is a trade-off, so every user needs to implement their own cache, which adds a burden on the user. We can move the implementation out of core-go now and add some implementations when users really request it.

Reply:
Moved `MemoryCache` to internal package `cachehelper` for unit test. Deleted `FileCache`